### PR TITLE
Add stake pool metadata to swagger.yaml

### PR DESCRIPTION
**Pull request by rvl**
Relates to #1063.

# Overview

Adds types, API JSON instances, and swagger spec for stake pool metadata.

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs [hunk 1]
```diff
@@ -518,6 +520,30 @@ data StakeDistribution = StakeDistribution
...
     } deriving (Eq, Show, Generic)
 
+-- | Information about a stake pool. This information is not used directly by
+-- cardano-wallet. It is sourced from the stake pool registry and passed
+-- straight through to API consumers.
+data StakePoolMetadata = StakePoolMetadata
+    { ticker :: StakePoolTicker
+    -- ^ Short human-readable ID for the stake pool.
+    , homepage :: Text
+    -- ^ Absolute URL for the stake pool's homepage link.
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Primitive/Types.hs]: **[suggestion: type-safety]**
Perhaps use [Network.URI](https://hackage.haskell.org/package/network-uri-2.6.1.0/docs/Network-URI.html) ?

**Why**
URIs have a quite well-defined format and we can rely on `parseURI` to parse the JSON string easily. It also makes it easier to then cross it with additional checks on the `uriScheme`.

File: lib/core/src/Cardano/Wallet/Api/Types.hs [hunk 2]
```diff
@@ -555,6 +556,22 @@ instance FromJSON ApiStakePoolMetrics where
...
+            <*> (v .: "pledge_address"))
+      where
+        parseTicker = eitherToParser . left ShowFmt . fromText
+
+instance ToJSON (ApiT StakePoolMetadata) where
+    toJSON (ApiT (StakePoolMetadata t h pa)) = object
+        [ "ticker" .= toText t
+        , "homepage" .= h
+        , "pledge_address" .= pa
+        ]
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Api/Types.hs]: **[suggestion: maintainability]** 

Use `Generic` to derive JSON instances. 

**Why**
1. Makes for more maintainable code in the long-run (avoid repetition of names and, make sure that the JSON encoding are consistent across many types).
2. More consistent with the style used across the rest of the module.

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs [hunk 3]
```diff
@@ -518,6 +520,30 @@ data StakeDistribution = StakeDistribution
     , unassigned :: Quantity "lovelace" Word64
     } deriving (Eq, Show, Generic)
 
+-- | Information about a stake pool. This information is not used directly by
+-- cardano-wallet. It is sourced from the stake pool registry and passed
+-- straight through to API consumers.
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Primitive/Types.hs]: :+1:

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs [hunk 4]
```diff
@@ -518,6 +520,30 @@ data StakeDistribution = StakeDistribution
...
+data StakePoolMetadata = StakePoolMetadata
+    { ticker :: StakePoolTicker
+    -- ^ Short human-readable ID for the stake pool.
+    , homepage :: Text
+    -- ^ Absolute URL for the stake pool's homepage link.
+    , pledgeAddress :: Text
+    -- ^ Bech32-encoded address.
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Primitive/Types.hs]: **[question]**
I wonder if we should keep a decoded representation here as an `Address`? That's a lot of encoding/decoding done for nothing however.

File: lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs [hunk 5]
```diff
@@ -441,6 +446,15 @@ spec = do
...
 
+        it "ApiT StakePoolMetadata" $ do
+            let msg = "Error in $: stake pool ticker length must be 3-4 characters"
+            Aeson.parseEither parseJSON [aesonQQ|
+                {
+                    "homepage": "https://12345",
+                    "ticker": "too long",
+                    "pledge_address": "ed25519_pk15vz9yc5c3upgze8tg5kd7kkzxqgqfxk5a3kudp22hdg0l2za00sq2ufkk7"
+                }
+            |] `shouldBe` (Left @String @(ApiT StakePoolMetadata) msg)
```
Comment (KtorZ) [inline lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs]: :+1:

**Review by KtorZ — APPROVED** [review]
Looks good, although some suggestions especially about the `Generic` part of the JSON types.

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs (hunk 1 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/Primitive/Types.hs]: I originally did use parseURI then removed it.
It doesn't add anything, unless we must sanitise URLs that users might click on.

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs (hunk 4 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/Primitive/Types.hs]: Exactly - there is no need to decode then re-encode the address.

File: lib/core/src/Cardano/Wallet/Api/Types.hs (hunk 2 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/Api/Types.hs]: OK done.

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs (hunk 1 above)
Comment (KtorZ) [reply lib/core/src/Cardano/Wallet/Primitive/Types.hs]: I think it's fine for _now_ as we'll only connect to the CF repository. Later, we may want to allow different sources, potentially "bad", to be chosen by users. 

Let's keep it simple for now.

**Pull request merged**